resources/js/components/Posts.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import Axios from "axios";
+import PostData from "./PostData";
+
+class Posts extends React.Component {
+    state = {
+        posts: [],
+        loading: true
+    };
+    fetchPosts = async () => {
+        const res = await Axios.get("/addPost");
+        if (res.data.status === 200) {
+            this.setState({ posts: res.data.posts });
+            this.setState({ loading: false });
+        }
+    };
+    componentDidMount() {
+        this.fetchPosts();
+    }
+
+    deletePost = async id => {
+        const res = await Axios.delete(`/addPost/${id}`);
+        if (res.data.status === 200) {
+            this.fetchPosts();
```

Comment:
you don't need to fetchPosts again, you can just remove it from state. If there are 1000's of posts it is extra load